### PR TITLE
UICIRC-627: UI tests replacement with RTL/Jest for component LostItemFeeSection (View)

### PR DESCRIPTION
**Pull request by annamelnyk**
## Purpose
UI tests replacement with RTL/Jest for `src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js`

## Refs
https://issues.folio.org/browse/UICIRC-627

## Screenshots
image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js [hunk 1]
```diff
@@ -44,8 +44,12 @@ const LostItemFeeSection = (props) => {
     : '-';
 
   return (
-    <div data-test-lost-item-policy-detail-fee-section>
+    <div
+      data-testid="viewLostItemFeeSectionTestId"
```
Comment (pavelspichonak) [inline src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js:48]: nit: Maybe let us not use postfix `TestId`. We already have `data-testid` attribute which says that it is test id

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  within,
+} from '@testing-library/react';
+
+import '../../../../../../test/jest/__mock__';
+
+import LostItemFeeSection from './LostItemFeeSection';
+
+const testSectionKeyValues = (testName, testId, expectedValue) => {
+  it(`should render ${testName} with ${expectedValue} value`, () => {
+    expect(screen.getByTestId(testId)).toBeVisible();
```
Comment (pavelspichonak) [inline src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js:15]: looks like we can store `screen.getByTestId(testId)` in variable to reuse it

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    beforeEach(() => {
+      render(<LostItemFeeSection {...lostItemFeeSectionProps} />);
+    });
+
+    it('should render LostItemFeeSection component', () => {
+      expect(screen.getByTestId('viewLostItemFeeSectionTestId')).toBeVisible();
+    });
+
+    it('should render open accordion with label', () => {
+      expect(screen.getByTestId('viewLostItemFeeSectionAccordionTestId')).toBeVisible();
```
Comment (pavelspichonak) [inline src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js:110]: let us store `screen.getByTestId('viewLostItemFeeSectionAccordionTestId')` to variable to reuse its value

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js (hunk 1 above)
Comment (annamelnyk) [reply src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js]: Good idea, agree, thanks

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js (hunk 2 above)
Comment (annamelnyk) [reply src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js]: Good idea, agree, thanks

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js [hunk 4]
```diff
@@ -112,7 +119,10 @@ const LostItemFeeSection = (props) => {
         </Row>
         <Row>
           <Col xs={12}>
-            <div data-test-lost-by-patron>
+            <div
+              data-testid="chargeAmountItemPatronTestId"
```
[the remark this answers is not in the document]
Comment (annamelnyk) [reply src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js]: Miss this case, thanks for noticing. Add tests.

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+describe('View LostItemFeeSection', () => {
+  const dash = '-';
+  const lostItemSectionAccordionLabel = 'ui-circulation.settings.lostItemFee.lostItemSection';
+  const displayYes = 'ui-circulation.settings.lostItemFee.yes';
+  const displayNo = 'ui-circulation.settings.lostItemFee.no';
+  const actualCost = 'ui-circulation.settings.lostItemFee.actualCost';
+  const lostItemReturnedCharged = 'ui-circulation.settings.lostItemFee.lostItemChargeOverdueBased';
+  const lostItemReturnedRemove = 'ui-circulation.settings.lostItemFee.lostItemRemoveOverdueFines';
+  const lostItemFeeSectionProps = {
```
[the remark this answers is not in the document]
Comment (annamelnyk) [reply src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js:30]: Maybe... don't think this is important.  This mock represents real data we usually receive from server

**Review by mkuklis — APPROVED** [review]

**Review by zburke — APPROVED** [review]

**Review by Dmitriy-Litvinenko — APPROVED** [review]

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js [hunk 6]
```diff
@@ -44,8 +44,12 @@ const LostItemFeeSection = (props) => {
...
-    <div data-test-lost-item-policy-detail-fee-section>
+    <div
+      data-testid="viewLostItemFeeSection"
+      data-test-lost-item-policy-detail-fee-section
+    >
       <Accordion
+        data-testid="viewLostItemFeeSectionAccordion"
         id="viewLostItemFeeSection"
         label={<FormattedMessage id="ui-circulation.settings.lostItemFee.lostItemSection" />}
         open={lostItemFeeSectionOpen}
```
Comment (Dmitriy-Litvinenko) [inline src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js:55]: Should we add test for section below (data-test-item-aged, data-test-patron-billed, data-test-item-recalled-aged, data-test-patron-billed-recalled, data-test-lost-item-fee, data-test-close-loan-after, data-test-replacement-fee)?

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+import '../../../../../../test/jest/__mock__';
+
+import LostItemFeeSection from './LostItemFeeSection';
+
+const testSectionKeyValues = (testName, testId, expectedValue) => {
+  it(`should render ${testName} with ${expectedValue} value`, () => {
+    const testIdValue = screen.getByTestId(testId);
+
+    expect(testIdValue).toBeVisible();
+    expect(within(testIdValue).getByText(expectedValue)).toBeVisible();
```
Comment (Dmitriy-Litvinenko) [inline src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js:18]: Can we add test for test label here? Just for example: expect(getItemByTestId('sectionTestId').getByText(labelSelector)).toBeVisible();

**Review by Dmitriy-Litvinenko — CHANGES_REQUESTED** [review]

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js (hunk 6 above)
Comment (annamelnyk) [reply src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.js:55]: Tests were added

File: src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js (hunk 7 above)
Comment (annamelnyk) [reply src/settings/LostItemFeePolicy/components/ViewSections/LostItemFeeSection/LostItemFeeSection.test.js]: Done

**Review by Stanislau-Snapok — APPROVED** [review]

**Pull request merged**